File: TiemposOrdenamiento/C++/insercion.cpp
#include<iostream>
#include <time.h>
#include <stdio.h>
#include <dos.h> 
#include<chrono>
using namespace std;

int main() {
	int i, n, *a;
	
	for (n=10; n<=500; n+=10){
	
		a = new int[n];
		
		//relleno
		for(i=0; i<n; i++){
			a[i] = i+1;
		}
		
		//ordenamiento
		int j, v;

		auto start = std::chrono::high_resolution_clock::now();
		for(i=1; i<n; i++){
			j = i;
			v = a[j];
			while (a[j-1]>v && j>0) {
				a[j] = a[j-1];
				j--;
			}
			a[j] = v;
		}
		auto finish = std::chrono::high_resolution_clock::now();
		cout << "N=" <<n<<", tiempo: "; 
	    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(finish-start).count() << "ns\n";
	    cout << endl;
	    
	}
	
	    
		//impresion
		for(i=0; i<n; i++){
			//cout<<a[i]<<",";
		}
		
		delete[] a;
}

File: TiemposOrdenamiento/Java/Ordenamiento/src/ordenamiento/Insercion.java
package ordenamiento;

public class Insercion {

	public static void main(String[] args) {

		int[] a;

		for (int n = 10; n <= 500; n += 10) {

			a = new int[n];

			// relleno
			for (int i = 0; i < n; i++) {
				a[i] = n - i;
			}

			// ordenamiento
			long t0 = System.nanoTime();
			for (int i = 1; i < n; i++) {
				int j = i;
				int v = a[j];
				while (j > 0 && a[j - 1] > v) { // comparacion
					a[j] = a[j - 1]; // 1/2 intercambio
					j--;
				}
				a[j] = v; // 1/2 inetercambio
			}
			long t1 = System.nanoTime();

			// impresion
			System.out.println("N=" + n + ", tiempo: " + (t1 - t0) + "ns");
			for (int i : a) {
				// System.out.print(i + ",");
			}
		}
	}

}
